public/pages/CreateIndexTemplate/containers/TemplateDetail/TemplateDetail.tsx

```diff
@@ -0,0 +1,599 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { forwardRef, useCallback, useContext, useEffect, useImperativeHandle, useRef, Ref, useState } from "react";
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiFormRowProps,
+  EuiLink,
+  EuiSpacer,
+  EuiTitle,
+} from "@elastic/eui";
+import flat from "flat";
+import { ContentPanel } from "../../../../components/ContentPanel";
+import AliasSelect from "../../../CreateIndex/components/AliasSelect";
+import IndexMapping, { IIndexMappingsRef, transformArrayToObject } from "../../../CreateIndex/components/IndexMapping";
+import { TemplateItem, TemplateItemRemote } from "../../../../../models/interfaces";
+import useField, { FieldInstance, transformNameToString } from "../../../../lib/field";
+import CustomFormRow from "../../../../components/CustomFormRow";
+import { AllBuiltInComponents } from "../../../../components/FormGenerator";
+import RemoteSelect from "../../../../components/RemoteSelect";
+import { ServicesContext } from "../../../../services";
+import { BrowserServices } from "../../../../models/interfaces";
+import AdvancedSettings from "../../../../components/AdvancedSettings";
+import { CoreServicesContext } from "../../../../components/core_services";
+import { CoreStart } from "opensearch-dashboards/public";
+import { submitTemplate, getTemplate } from "./hooks";
+import DescriptionListHoz from "../../../../components/DescriptionListHoz";
+import { Modal } from "../../../../components/Modal";
+import JSONEditor from "../../../../components/JSONEditor";
+import { RouteComponentProps } from "react-router-dom";
+import { INDEX_SETTINGS_URL, ROUTES } from "../../../../utils/constants";
+import DeleteTemplateModal from "../../../Templates/containers/DeleteTemplatesModal";
+import TemplateType, { TemplateConvert } from "../../components/TemplateType";
+import { filterByMinimatch } from "../../../../../utils/helper";
+
+export interface TemplateDetailProps {
+  templateName?: string;
+  onCancel?: () => void;
+  onSubmitSuccess?: (templateName: string) => void;
+  readonly?: boolean;
+  history: RouteComponentProps["history"];
+}
+
+const TemplateDetail = ({ templateName, onCancel, onSubmitSuccess, readonly, history }: TemplateDetailProps, ref: Ref<FieldInstance>) => {
```

Comment:
Let's split this into smaller components placed in separate files. Easier to review and maintain

Reply:
Fixed on https://github.com/opensearch-project/index-management-dashboards-plugin/commit/a57c86571fefe2f218d46776b1f2dc623d922449